native/logik_simulation/src/ffi/mod.rs

```diff
@@ -54,16 +36,40 @@ pub extern "C" fn remove_subnet(data: *mut Data, id: i32) -> bool {
 pub extern "C" fn add_component(data: *mut Data, component: i32) -> i32 {
     let data = unsafe { &mut *data};
     
-    let component: Box<dyn Component> = match component {
-        0 => Box::new(Output {}),
-        _ => Box::new(AND {}) // TODO: make me the same as the ID:s in C#
+    let comp: Box<dyn Component> = match component {
+        1 => Box::new(Constant::new()),
+        2 => Box::new(OutputGate {}),
+        3 => Box::new(InputGate {}),
+        5 => Box::new(LED {}),
+        8 => Box::new(Button::new()),
+        50 => Box::new(Buffer {}),
+        51 => Box::new(NOT {}),
+        52 => Box::new(AND {}),
+        53 => Box::new(NAND {}),
+        54 => Box::new(OR {}),
+        55 => Box::new(NOR {}),
+        56 => Box::new(XOR {}),
+        57 => Box::new(XNOR {}),
+        60 => Box::new(TriBuffer {}),
+        61 => Box::new(TriInverter {}),
+        100 => Box::new(DFlipFlop::new()),
+        101 => Box::new(TFlipFlop::new()),
+        102 => Box::new(JKFlipFlop::new()),
+        103 => Box::new(SRFlipFlop::new()),
+        300 => Box::new(Probe {}),
+        302 => Box::new(Clock::new()),
+        _ => unreachable!()
     };
     
-    let p = component.ports();
+    let p = comp.ports();
+    
+    let res = data.add_component(comp, iter::repeat(None).take(p).collect()).unwrap();
     
-    let res = data.add_component(component, iter::repeat(None).take(p).collect());
+    if component == 302 {
```

Comment:
See earlier comment about enum (or just named constants).

Reply:
Resolved in 0ef1a4e